### PR TITLE
[WiP]Сreate UTs for AppIconsFolder

### PR DESCRIPTION
**Pull request by okozlovlux**
Created UTs for conditions for SDL to create and use 'AppIconsFolder' storage

Related to [APPLINK-29291](https://adc.luxoft.com/jira/browse/APPLINK-29291)

### CODE REVIEW

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
@LevchenkoS @LuxoftAKutsan @dcherniev @indlu @dtrunov please review

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
@LevchenkoS @LuxoftAKutsan @dcherniev please review

File: src/components/config_profile/test/profile_test.cc [hunk 1]
```diff
@@ -724,6 +724,22 @@ TEST_F(ProfileTest, CheckIntContainerInSecurityData) {
...
+            profile_.app_icons_folder());
+  EXPECT_EQ(104857611u, profile_.app_icons_folder_max_size());
+}
+
+TEST_F(ProfileTest, AppIconsFolder_DefaultValues_SUCCESS) {
+  profile_.set_config_file_name("smartDeviceLink.ini");
+  EXPECT_EQ("smartDeviceLink.ini", profile_.config_file_name());
+  EXPECT_EQ(file_system::CurrentWorkingDirectory(),
+            profile_.app_icons_folder());
+  EXPECT_EQ(104857600u, profile_.app_icons_folder_max_size());
```
Comment (dcherniev) [inline src/components/config_profile/test/profile_test.cc]: @okozlovlux This value is mismatching with the value in smartDeviceLink.ini file (AppIconsFolderMaxSize = 104857500), shouldn't it be the same?

File: src/components/application_manager/test/application_manager/application_manager_test.cc [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    EXPECT_CALL(mock_application_manager_settings_, stop_streaming_timeout())
+        .WillRepeatedly(Return(kStopStreamingTimeout));
+    EXPECT_CALL(mock_application_manager_settings_,
+                app_resumption_save_persistent_data_timeout())
+        .WillOnce(ReturnRef(kAppId));
+    EXPECT_CALL(mock_application_manager_settings_, max_number_of_ios_device())
+        .WillRepeatedly(Return(0));
+    EXPECT_CALL(mock_policy_settings_, enable_policy())
+        .WillOnce(Return(true))
+        .WillOnce(Return(false));
```
Comment (dcherniev) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @okozlovlux I haven't seen such duplication of WillOnce() statement, is this expectation correct?

File: src/components/config_profile/test/profile_test.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/config_profile/test/profile_test.cc]: @dcherniev , yes. Value from ini file is less than default value (`104857600`).  In that case we use default

File: src/components/application_manager/test/application_manager/application_manager_test.cc (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @dcherniev , yes. it is correct

File: src/components/config_profile/test/profile_test.cc (hunk 1 above)
Comment (dcherniev) [inline src/components/config_profile/test/profile_test.cc]: @okozlovlux Ok

File: src/components/application_manager/test/application_manager/application_manager_test.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @okozlovlux Ok

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
Rebuild required

File: src/components/application_manager/test/application_manager/application_manager_test.cc [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        .WillRepeatedly(Return(false));
+    EXPECT_CALL(mock_application_manager_settings_, launch_hmi())
+        .WillOnce(Return(kHmiLaunch));
+    EXPECT_CALL(mock_application_manager_settings_, stop_streaming_timeout())
+        .WillRepeatedly(Return(kStopStreamingTimeout));
+    EXPECT_CALL(mock_application_manager_settings_,
+                app_resumption_save_persistent_data_timeout())
+        .WillOnce(ReturnRef(kAppId));
+    EXPECT_CALL(mock_application_manager_settings_, max_number_of_ios_device())
+        .WillRepeatedly(Return(0));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @okozlovlux, are all of this calls are really should be expected? May we use "ON_CALL" for this?

Comment (okozlovlux) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @LevchenkoS , fixed 2bbf434

**Comment by okozlovlux** [comment]
@LuxoftAKutsan please review

**Comment by okozlovlux** [comment]
Rebuild required

File: src/components/application_manager/test/application_manager/application_manager_test.cc [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        .WillByDefault(ReturnRef(kAppId));
+    ON_CALL(mock_application_manager_settings_, max_number_of_ios_device())
+        .WillByDefault(Return(0));
+    EXPECT_CALL(mock_policy_settings_, enable_policy())
+        .WillOnce(Return(true))
+        .WillOnce(Return(false));
+
+    file_system::CreateDirectory(kAppStorageFolder);
+
+    app_manager_impl_ = new am::ApplicationManagerImpl(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @okozlovlux raw pointer?

File: src/components/application_manager/test/application_manager/application_manager_test.cc [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        mock_application_manager_settings_, mock_policy_settings_);
+    last_state_ = new resumption::LastState(kAppStorageFolder, kAppInfoStorage);
+  }
+
+  void TearDown() OVERRIDE {
+    delete app_manager_impl_;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @okozlovlux can you shared pointers or auto pointer or values?

File: src/components/config_profile/test/profile_test.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/config_profile/test/profile_test.cc]: @okozlovlux please extract and descripe this magic number

File: src/components/application_manager/test/application_manager/application_manager_test.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @LuxoftAKutsan , fixed d9707ed

File: src/components/application_manager/test/application_manager/application_manager_test.cc (hunk 5 above)
Comment (okozlovlux) [inline src/components/application_manager/test/application_manager/application_manager_test.cc]: @LuxoftAKutsan , fixed d9707ed

File: src/components/config_profile/test/profile_test.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/config_profile/test/profile_test.cc]: @LuxoftAKutsan , fixed d9707ed

**Comment by okozlovlux** [comment]
Rebuild required

**Pull request merged**